test/reflection.jl

```diff
@@ -213,3 +213,10 @@ let t13464 = "hey there sailor"
         @test startswith(err13464.msg, "expression is not a function call, or is too complex")
     end
 end
+
+# Issue 14173
+module Tmp
+    export A
+    A = randn(2000, 2000)
+end
+@test @elapsed(whos(IOBuffer(), Tmp)) < 5
```

Comment:
Good idea with the length of the string, but I've just realized that the string buffer is truncated to the screen width after it has been populated with all the elements of the array.

Another problem is that `showlimited` shows the type information in a header whereas `show` doesn't. Therefore with my solution, the type information is repeated, e.g.
```jl
julia> whos(STDOUT, Tmp14173)
                             A  31250 KB     2000x2000 Array{Float64,2} : 2000x2000 Array{Float64,2}:…
                      Tmp14173  31250 KB     Module : Tmp14173
```
I'll look at this again tomorrow and see if I can come up with a better solution.